src/compute/src/compute_state.rs

```diff
@@ -190,20 +190,10 @@ impl<'a, A: Allocate, B: ComputeReplay> ActiveComputeState<'a, A, B> {
                     self.compute_state.pending_peeks.push(peek);
                 }
             }
-            ComputeCommand::CancelPeeks { uuids } => {
-                let pending_peeks_len = self.compute_state.pending_peeks.len();
-                let mut pending_peeks = std::mem::replace(
-                    &mut self.compute_state.pending_peeks,
-                    Vec::with_capacity(pending_peeks_len),
-                );
-                for peek in pending_peeks.drain(..) {
-                    if uuids.contains(&peek.peek.uuid) {
-                        self.send_peek_response(peek, PeekResponse::Canceled);
-                    } else {
-                        self.compute_state.pending_peeks.push(peek);
-                    }
-                }
-            }
+            ComputeCommand::CancelPeeks { uuids } => self
+                .compute_state
+                .pending_peeks
+                .retain(|peek| !uuids.contains(&peek.peek.uuid)),
```

Comment:
I think this change is not correct, in that sending a peek response of `Canceled` is how we communicate back to the surrounding compute infrastructure that the peek has been dealt with. Not doing that will result I believe in the compute instance retaining the peek indefinitely, and the dataflow on which it depends, and redeploying them to all replicas.